src/main/java/com/liveramp/daemon_lib/DaemonCallback.java

```diff
@@ -0,0 +1,17 @@
+package com.liveramp.daemon_lib;
+
+import com.liveramp.daemon_lib.utils.DaemonException;
+
+public interface DaemonCallback {
```

Comment:
How about just `public interface DaemonCallback extends Callable`?

Reply:
Yeah I think that is cleaner. We could also just get rid of this interface completely and have the `wakeUpCallback` be of type `Callable<Void>`, but I think I still like having this interface as a way of saying that a callback in this daemon _is_ a callable with type Void, since the daemon is doing nothing with the value that the call returns

Reply:
I think you're right in that we should keep the interface. Should it perhaps be `Runnable`, by the way?

Reply:
You're right. Runnable makes more sense here if it is just returning void. Will change

Reply:
Though one thing to consider is that Runnable's can't throw exceptions. So I guess it depends on if we want people who are writing wakeup callbacks to handle their own checked exceptions or not. I think it is reasonable to have implementers of the wakeup callback handle their checked exceptions, so I think Runnable probably makes sense here.

Reply:
I think your reasoning is sound here.